src/t8_forest/t8_forest_vtk.cxx

```diff
@@ -404,7 +406,10 @@ for (int idata = 0; idata < num_data; idata++) {
 /* We set the input data and write the vtu files. */
 pwriterObj->SetInputData (unstructuredGrid);
 pwriterObj->Update ();
-pwriterObj->Write ();
+if (pwriterObj->Write () == 0) {
+  /* Writing failed */
+  goto t8_forest_vtk_failure;
```

Comment:
The clean up is the same even if writing was successful, so the code is currently just duplicated, which we should avoid.

What you could do is store 1 or 0 in a variable depending on the result here, then do the clean up and return the variable.